### DISCUSSION TITLE
Diversidade e acessibilidade devem ser sempre atendidas ao desenvolver um software?

### DISCUSSION THREAD

**Discussion by fabiomirandaa** [frontendbr/forum#1846] [Discussão]
Queria iniciar uma discussão sobre 2 pontos. Quando vamos decidir se damos suporte ou não a determinado navegador se vamos criar um novo sistema pra substituir um Sistema Legado, geralmente consultamos métricas de uso. Se a porcentagem de uso do navegador for bem baixa, geralmente deixamos de dar o suporte. 

Agora, em questão de UX e Acessibilidade, como deveria ser o modelo ideal? 

Eu pessoalmente penso que se, por exemplo, 0.5% dos usuários do sistema são deficientes visuais, o software deve estar com recursos de acessibilidade(realmente, não para inglês ver), porque nesse caso mais do que números, são pessoas que ficariam impossibilitadas de usar o software. Confesso que eu não tenho aplicado tanto essa minha visão na prática, mas sempre penso que deveria.

Agora questão de diversidade, não sei como seria mensurável. Mas, para entender o que quero dizer: Digamos que um software seja feito e que a maior parte das pessoas que acessam seja entre 15 e 35 anos. Mas, eu devo levar em consideração se o uso é bom para idosos? Devo levar em consideração se pessoas com pouca instrução, ainda que jovens conseguirão usar? Como fazer para fazer esse tipo de teste de usabilidade?

**Comment by ninetails** [comment]
Jogarei a realidade que sinto dos lugares em que trabalhei. Isso deveria ser uma diretriz da empresa.

Só que vamos pegar o atual posicionamento no Brasil: o único lugar que eu ouço falar sobre é em forums da W3C/Nic.br e alguns poucos meetups e isso é preocupante. Estou longe de ser expert no assunto, mas mesmo no campo legal [não há muita coisa](https://www.acessoparatodos.com.br/acessibilidadeweb.php) e enquanto não houver ações legais polêmicas estilo o da [Domino's ano passado](https://gizmodo.uol.com.br/dominos-processada-site-pessoas-deficiencia/) pra cá, acho difícil mudar.

Dado isso, creio que atualmente a melhor forma de testar seria contratando uma consultoria especializada nisso, geralmente eles já possuem grupos para testes com tais perfis. E diria que se a empresa chega a pensar nisso, penso eu que teria grana pra bancar tal consultoria já que são poucas as que se preocupam.

Ou seja, saindo disso fica a cargo de nós desenvolvedores. Existem ferramentas que auxiliam em testes automatizados mas creio que boa parte seria nos _achometros_ :disappointed: . Levanto algumas: [axe](https://www.deque.com/axe/) e o [ChromeVox](https://chrome.google.com/webstore/detail/chromevox-classic-extensi/kgejglhpjiefppelpmljglcjbhoiplfn?hl=en) se não tiver um leitor de tela, mas existem muitas outras...

Alguém que manja pra caramba e que indico seguir é o [Marcelo Sales](https://twitter.com/msales).

**Comment by lddv** [comment]
Também vou deixar a recomendação aqui de outro profissional da área de acessibilidade: o [Thiago](https://github.com/althi) - [Twitter dele](https://twitter.com/althi).

Entendo muito do que você quer sobre diversidade seria coberto pela acessibilidade, pois a aplicação acessível pode suportar deficiências transitórias, como estar com um braço quebrado e precisar de não usar o mouse por algum tempo. Mesmo uma fonte que está muito pequena para uma pessoa idosa com problema de visão pode ser visto como acessibilidade.

A questão de diversidade que você mencionou também pode entrar na questão de conteúdo e engajamento dos usuários. Seria uma outra área de foco, na minha opinião.
